### DISCUSSION TITLE
defaultTranslate for FormField

### DISCUSSION THREAD

**Discussion by korenevskiy** [joomla/joomla-cms#35833] [General]
I kind of saw one (two) months ago, the `FormField` fields have the `defaultTranslate` property.
Now I'm looking and I don't see him.
Have we deleted it? Or did I think I saw him?
:-(

**Comment by korenevskiy** [comment]
Can I do PR? By creating the `defaultTranslate` property for the `FormField` field?

**Reply by brianteeman** [reply]
What does it do?

But in general no one needs to ask for permission to create a pr

**Reply by korenevskiy** [reply]
Yes, of course. But I'm just confused. I kind of saw support for default Translate in the repository. And now I don't see it. And I'm probably asking if I'm having phantom hallucinations or if it really happened, and then these improvements were canceled.?

Suppose a plugin or module has a control drop-down list with a list of currencies.
And depending on the localization, the default value will be selected according to the principle of language localization.
And it can also be a field with an invitation. The default value of which will be selected depending on the localization language.

**Reply by brianteeman** [reply]
the only thing I can think of is I think if you add translate=true in the field definition then the values in the option will be translated (if a translation exists)

check how that is used in core it may be what you are looking for

**Reply by korenevskiy** [reply]
No, it doesn't work. But I also didn't find such an attribute in the code at all.

**Reply by brianteeman** [reply]
Not sure where you are looking it is used a lot

https://github.com/joomla/joomla-cms/search?l=PHP&q=%22%24translate%22

**Comment by korenevskiy** [comment]
Oh!
https://github.com/joomla/joomla-cms/blob/bb56578b0151549cc984863ec08510a79836ad6e/libraries/src/Form/Form.php#L1624
It's cruel to mock people like that.
This code should be located inside the Form Field class.
No wonder I couldn't find this property.
This is a big mistake. He shouldn't be there.
There is a desire, inside the Form Field class, to write a comment about the support of the attribute `translate_default`.
There is a suspicion that I am still healthy.

**Reply by brianteeman** [reply]
I am not a developer I am just trying to help answer your question by pointing you in a possible direction that could help

**Reply by korenevskiy** [reply]
Your help was very useful to me in any case. Communicating with you, I found the answer. I think this is due to the fact that you really wanted to help. I appreciate it anyway.
And I am very glad that I did not have visions and hallucinations.

**Reply by brianteeman** [reply]
Glad you worked out how to do what you needed to do
